tests/bears/DebugBearsTest.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import sys
+
+from io import StringIO
+
+from coalib.bears.Bear import Debugger, debug_run
+
+
+def func1(*args, **kwargs):
+    yield 1
+    yield 2
+    yield 3
+
+
+def func2(*args, **kwargs):
+    return([1, 2])
+
+
+def func3(*args, **kwargs):
+    kwargs['Makman'] = 'Admin'
+    return func1(*args, **kwargs)
+
+
+class DebugBearsTest(unittest.TestCase):
+    def setUp(self):
+        self.trace = sys.gettrace()
+        self.output = StringIO()
+
+    def tearDown(self):
+        sys.settrace(self.trace)
+
+    def test_debug_run_yield(self):
+        args = ('a', ('b', 'c'))
+        kwargs = {'d': 'e'}
+        input = StringIO('q\nc\nq\nc')
+        dbg = Debugger(stdin=input, stdout=self.output)
+        self.assertEqual(debug_run(func1, dbg, *args, **kwargs), [1, 2, 3])
+        output = self.output.getvalue()
+        # restore the coverage settrace to prevent the coverage breakage
+        # on project because  we can't chain coverage trace to run parallel
+        # with debugger. To increase the coverage Mock test has been added in
+        # BearTest file.
+        self.assertEqual(output.split('\n')[1], '-> yield 1')
+        self.assertEqual(output.split('\n')[3], '-> yield 2')
+        self.assertEqual(output.split('\n')[5], '-> yield 3')
+
+    def test_debug_run_list(self):
+        args = ('a', ('b', 'c'))
+        kwargs = {'d': 'e'}
+        input = StringIO('q\nc')
+        dbg = Debugger(stdin=input, stdout=self.output)
+        self.assertEqual(debug_run(func2, dbg, *args, **kwargs), [1, 2])
+        output = self.output.getvalue()
+        self.assertEqual(output.split('\n')[1], '-> return([1, 2])')
+
+    def test_debug_run_generator(self):
```

Comment:
this name sounds technically equal to `test_debug_run_yield`. Can you call it differently / more precise?